src/library/scala/collection/immutable/List.scala

```diff
@@ -134,7 +147,26 @@ sealed abstract class List[+A]
   // When calling prependAll with another list `prefix`, avoid copying `this`
   override def prependedAll[B >: A](prefix: collection.IterableOnce[B]): List[B] = prefix match {
     case xs: List[B] => xs ::: this
-    case _ => super.prependedAll(prefix)
+    case _ =>
+      if (isEmpty) {
+        List.from(prefix)
+      } else {
```

Comment:
How about doing it the other way around? `List.from` could call `Nil.prependedAll` to avoid duplicating the implementation. The performance impact should be minimal.

Reply:
Tried it, results here: https://gist.github.com/joshlemer/9839b185a088dbce7285a163ec0d70ce

Reply:
The results look quite noisy so I reran it on one of my benchmarking machines. It's still not really conclusive:

```
[info] Benchmark                     (size)  Mode  Cnt      Score      Error  Units
[info] ListBenchmark.fromVectorJosh       0  avgt   40      3.968 ±    0.527  ns/op
[info] ListBenchmark.fromVectorJosh       1  avgt   40     14.561 ±    0.055  ns/op
[info] ListBenchmark.fromVectorJosh      10  avgt   40     46.556 ±    0.089  ns/op
[info] ListBenchmark.fromVectorJosh      50  avgt   40    280.689 ±   18.161  ns/op
[info] ListBenchmark.fromVectorJosh     100  avgt   40    423.795 ±    3.993  ns/op
[info] ListBenchmark.fromVectorJosh    1000  avgt   40   3708.043 ±    6.685  ns/op
[info] ListBenchmark.fromVectorJosh   10000  avgt   40  44213.077 ± 4140.386  ns/op
[info] ListBenchmark.fromVectorStef       0  avgt   40      3.887 ±    0.571  ns/op
[info] ListBenchmark.fromVectorStef       1  avgt   40     15.278 ±    0.140  ns/op
[info] ListBenchmark.fromVectorStef      10  avgt   40     58.610 ±    0.082  ns/op
[info] ListBenchmark.fromVectorStef      50  avgt   40    275.619 ±   21.364  ns/op
[info] ListBenchmark.fromVectorStef     100  avgt   40    497.496 ±   45.946  ns/op
[info] ListBenchmark.fromVectorStef    1000  avgt   40   3594.043 ±   10.796  ns/op
[info] ListBenchmark.fromVectorStef   10000  avgt   40  43906.390 ± 4627.277  ns/op
```